src/removeHighlightWrappers.ts

```diff
@@ -12,52 +12,59 @@ const NODE_TYPE = {
  * Removes highlights from element. If element is a highlight itself, it is removed as well.
  * If no element is given, all highlights all removed.
  * @param {HTMLElement} [element] - element to remove highlights from
+ * @param {HTMLElement} [filterFn] - filter which highlights should be removed from the element, default: all
  */
-export function removeAllHighlights(element: HTMLElement) {
-  getHighlights(element).forEach(removeHighlightElement);
+export function removeAllHighlights(element: HTMLElement, filterFn?: (element: HTMLElement) => boolean) {
+  getHighlights(element).forEach((element) => removeHighlightElement(element, filterFn));
 }
 
-export default function(highlight: Highlight) {
-  highlight.elements.forEach(removeHighlightElement);
+export default function(highlight: Highlight, filterFn?: (element: HTMLElement) => boolean) {
+  highlight.elements.forEach((element) => removeHighlightElement(element, filterFn));
```

Comment:
`highlight.elements` is the span elements for this highlight, it seems like you are trying to refine the purpose of the `removeHighlightElement` function to only remove elements belonging to a single highlight model, but within that function it does `getHighlights`. since we already know the elements belonging to this model from `highlight.elements`, the call to `getHighlights` could only possibly return elements belonging to different models, which you then have to filter out.

i feel like this can be simplified. the `removeHighlightElement` could simply unwrap its `element` and then merge the resulting siblings. the querying, filtering, and looping can all probably be removed without affecting any functionality

Reply:
In `Highlighter.erase()` I'm passing a filter function that will cause `removeHighlightElement` to remove elements only from the current highlighting model, but in the `XpathRangeSelector/index.ts` I'm passing a filter function that will remove only highlights that are not matching current highlight model and in the `removeAllHighlights` I'll not pass a filter function so all highlights will be removed.

I think that maybe if I refactor somehow a change in `XpathRangeSelector/index.ts` so it doesn't have to remove highlights I could then also refactor this logic

any thoughts?

Reply:
when calling this default function, `highlight.elements` will be literally the list of spans associated with this model, so there is no need to query anything you can just unwrap them individually and then merge siblings

when calling `removeAllHighlights`, such as from rangeContents, you want to remove any highlight spans, which is what `getHighlights` returns, so no querying is needed there either

if you wanted to add a filter to `removeAllHighlights`, you could do it on the return value of `getHighlights`, before passing them into `removeHighlightElement`, there is still no querying needed within `removeHighlightElement`

as i said in my other comment, i think that it is destructive and unwanted to call removeHighlight in `src/serializationStrategies/XpathRangeSelector/index.ts`, because you're removing highlights that were previously added to the page and not putting them back again, so anything that was there before will be gone. that logic in there is supposed to work with other highlights in the way, so there might be a bug in that